src/Forms/Form.php

```diff
@@ -1480,7 +1479,7 @@ public function loadDataFrom($data, $mergeStrategy = 0, $fieldList = null)
      * Save the contents of this form into the given data object.
      * It will make use of setCastedField() to do this.
      *
-     * @param DataObjectInterface $dataObject The object to save data into
+     * @param ViewableData&DataObjectInterface $dataObject The object to save data into
```

Comment:
It must be both. It must implement `DataObjectInterface` because those methods are explicitly called. It must be `ViewableData` because we don't support people throwing random objects at the wall. There's things like `hasMethod` that we want to be able to rely on being there.